### ISSUE TITLE
mask-rcnn edge agreement giving poor results as compared to original mask rcnn

### ISSUE THREAD

**Issue by supritimulay** [FlashTek/mask-rcnn-edge-agreement-loss#2]
Hi,

I was trying to run mask rcnn edge agreement on my own dataset and tried to segment with this but results are not as expected. I also ran the original Mask rcnn on these images and results are ok except at edges and tricky corners.  I train the network with edge agreement as  python3 liver_CT.py train --dataset=/Mask_RCNN/mask-rcnn-edge-agreement-loss-master/dataset/Liver_CT_YIQ/ --weights=imagenet. Same as the original mask rcnn expect using new codes. 

Can you help me to know whether I have trained the data correctly or I need to do some more changes for training on your own data.

Thanks,
Supriti

**Comment by JulienSiems** [comment]
Hey! 

Could you give us more details on your data and the ground truth annotations? It’s really hard to judge what might be going wrong without knowing more about what Mask R-CNN will have to predict in this case.

**Comment by supritimulay** [comment]
Data is the CT image slices converted to png images and ground truth is image masks(binary images) in png which is converted to json file as per mask rcnn requirement. On these images mask rcnn segementation is nearly accurate except for certain boundaries and sharp corners. But mask rcnn with edge agreement not giving the better result. In fact its segmenting region lesser then mask rcnn.

**Comment by JulienSiems** [comment]
In this case the quality of the instance segmentation mask edges might have an impact. Are they mostly coarse or do they follow visible image boundaries closely? 
Could you post a screenshot of the edge agreement loss depicted over some epochs? Does it go down?

**Comment by supritimulay** [comment]
Hi,

Please find attached segmented output from original Mask RCNN and with edge
agreement loss.
I am doing any mistake while training or any parameters need to be changed?

Thanks,
Supriti

**Comment by supritimulay** [comment]
Hi, 
I have sent the predicted images through mail for original Mask RCNN and with edge agreement loss.
I am doing any mistake while training edge agreement or any parameters need to be changed?
I ran both the networks with 30 epochs.

**Comment by zimmerrol** [comment]
@supriti-m We did not receive an e-mail from you, but that's no problem at all: Please do _not_ send us any files but rather post all the information here in the issue so that people who are having similar issues can follow our discussion here and get all the information that was eventually needed to solve it.

The Edge Agreement Loss has already been used in different projects, too, where it was used to refine the predictions on biomedical imaging data; therefore, I do not see any reason why it should fail on your data in general. I assume that it is just a problem with your configuration. To check this we need to see exactly how you start the training procedure and also what other modifications in the code you have done.

**Comment by supritimulay** [comment]
I will attach the files here as well. I have used balloon model as my base and added the edge agreement loss functionality in balloon.py as I am segmenting only 1 class.  I use balloon inspect model for predicting the class and used edge agreement loss  functions accordingly. 
Please let me know if I am wrong somewhere.
![edge_agrrement_pred](https://user-images.githubusercontent.com/22741931/59750930-3c422780-929d-11e9-8118-8d8988c6b34a.png)
![original_mask_rcnn](https://user-images.githubusercontent.com/22741931/59750931-3c422780-929d-11e9-8733-1895625b8a90.png)

**Comment by JulienSiems** [comment]
Could you be more precise? Did you use our implementation of the Edge Agreemen Loss? Did you make sure to add it to the losses that are used for optimization? You could check that by inspecting the gradients. 

Also please annotate the pictures in the previous post to indicate what the picture is showing (Prediction with Edge Agreement Loss etc.). Please also show us an example of the groundtruth annotation. Again it would be helpful to see the progression of the edge agreement loss during training so please post a graph for example the one we write to tensorboard.

**Comment by supritimulay** [comment]
I have taken implementation of edge agreement loss and trained the data with default parameters.  In the above images first image is with edge agreement  and below image is with original mask rcnn. I have sent the mail to you in detail. Thanks for your help on the same.

**Comment by zimmerrol** [comment]
As there is no new information about the issue I will close it.
